src/middlewared/middlewared/plugins/kubernetes_linux/cni.py

```diff
@@ -0,0 +1,114 @@
+import asyncio
+import ipaddress
+import subprocess
+
+from middlewared.plugins.interface.netif import netif
+from middlewared.service import CallError, ConfigService
+
+from .k8s import api_client, service_accounts
+
+
+class KubernetesCNIService(ConfigService):
+
+    class Config:
+        private = True
+        namespace = 'k8s.cni'
+
+    async def config(self):
+        return {
+            'multus': {'service_account': 'multus'},
+            'kube_router': {'service_account': 'kube-router'},
+        }
+
+    async def setup_cni(self):
+        kube_config = await self.middleware.call('datastore.query', 'services.kubernetes', [], {'get': True})
+        config = await self.config()
+        async with api_client() as (api, context):
+            cni_config = kube_config['cni_config']
+            for cni in config:
+                if not await self.validate_cni_integrity(cni, kube_config):
+                    cni_config[cni] = await service_accounts.get_service_account_details(
+                        context['core_api'], config[cni]['service_account']
+                    )
+
+        await self.middleware.call(
+            'datastore.update', 'services.kubernetes', kube_config['id'], {'cni_config': cni_config}
+        )
+        await self.middleware.call('etc.generate', 'cni')
+        await self.middleware.call('service.start', 'kuberouter')
+        await asyncio.sleep(5)
```

Comment:
Is this the best way to wait for routing tables to appear?

Reply:
You could just do a call to the docker API and wait while that call fails.